src/k8s/cmd/k8s/k8s_join_cluster.go

```
@@ -78,6 +78,9 @@ func newJoinClusterCmd(env cmdutil.ExecutionEnvironment) *cobra.Command {
 			cmd.PrintErrln("Joining the cluster. This may take a few seconds, please wait.")
 			if err := client.JoinCluster(cmd.Context(), apiv1.JoinClusterRequest{Name: opts.name, Address: opts.address, Token: token, Config: joinClusterConfig}); err != nil {
 				cmd.PrintErrf("Error: Failed to join the cluster using the provided token.\n\nThe error was: %v\n", err)
+				cmd.Printf("Cleaning up...\n")
+				client.RemoveNode(cmd.Context(), apiv1.RemoveNodeRequest{Name: opts.name, Force: true})
+				client.DeleteClusterJoin(cmd.Context(), apiv1.ClusterJoinCleanupRequest{})
```

same here, do not ignore errors, print them as warnings